.docker/build/build.sh

```diff
@@ -185,8 +185,16 @@ function refresh_hints() {
     # Silent, always-successful merge
     export GIT_MERGE_AUTOEDIT=no
     git merge $commit -Xtheirs
+
+    # If build hints branch exists on remote, remove it
+    exists=$(git branch -r -l "origin/BuildHints-$CI_BRANCH")
+    if [ ! -z $exists ]; then
+        git push $remote :BuildHints-$CI_BRANCH
+    fi
+
     # Push.
-    git push $remote $CI_BRANCH
+    git checkout -b BuildHints-$CI_BRANCH
+    git push $remote BuildHints-$CI_BRANCH
```

Comment:
This change will eventually make it into `dev` and will trigger a hints refresh for every commit.
I think this is ok, but it changes the status quo of refreshing hints automatically only in `fstar-master`.

Reply:
@protz reminded that this only happens for scheduled builds (the `hacl-nightly`) target. 
So, unless we schedule builds of `dev` hints won't be refreshed automatically. If this is still a problem, we could easily modify the build logic to exclude `dev`.
This also means that we'll be able to trigger an automatic hint refresh in `dev` by manually queueing a `hacl-nightly` build from VSTS, which I think it's a nice feature.

I think this change should stand; sorry for the noise.